libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/SignupGoogleFragment.java

```diff
@@ -186,34 +224,49 @@ public void onSocialChanged(OnSocialChanged event) {
                     mLoginListener.showSignupToLoginMessage();
                     // Dispatch social login action to retrieve data required for two-factor authentication.
                     PushSocialPayload payload = new PushSocialPayload(mIdToken, SERVICE_TYPE_GOOGLE);
+                    AppLog.d(T.MAIN,
+                            "GOOGLE SIGNUP: onSocialChanged - error - two step authentication - dispatching "
+                            + "pushSocialLoginAction");
                     mDispatcher.dispatch(AccountActionBuilder.newPushSocialLoginAction(payload));
                     break;
                 // WordPress account exists with input email address, but not connected.
                 case USER_EXISTS:
-                    mAnalyticsListener.trackSignupSocialAccountsNeedConnecting();
-                    mAnalyticsListener.trackSignupSocialToLogin();
-                    mLoginListener.showSignupToLoginMessage();
-                    mLoginListener.loginViaSocialAccount(mGoogleEmail, mIdToken, SERVICE_TYPE_GOOGLE, true);
-                    // Kill connections with FluxC and this fragment since the flow is changing to login.
-                    mDispatcher.unregister(this);
-                    getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - user already exists");
+                    handleUserExists();
+                    break;
+                // Too many attempts on sending SMS verification code. The user has to wait before they try again
+                case SMS_CODE_THROTTLED:
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - sms code throttled");
+                    showError(getString(R.string.login_error_sms_throttled));
                     break;
                 default:
-                    showErrorDialog(getString(R.string.login_error_generic));
+                    AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - error - unknown");
+                    showError(getString(R.string.login_error_generic));
                     break;
             }
             // Response does not return error when two-factor authentication is required.
-        } else if (event.requiresTwoStepAuth) {
+        } else if (event.requiresTwoStepAuth || "sms".equals(event.notificationSent)) {
+            AppLog.d(T.MAIN, "GOOGLE SIGNUP: onSocialChanged - 2fa required");
             mAnalyticsListener.trackSignupSocialToLogin();
             mLoginListener.needs2faSocial(mGoogleEmail, event.userId, event.nonceAuthenticator, event.nonceBackup,
                     event.nonceSms);
-            // Kill connections with FluxC and this fragment since the flow is changing to login.
-            mDispatcher.unregister(this);
-            getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
+            finishSignUp();
+        } else {
```

Comment:
We shouldn't make the assumption all scenarios that aren't an error and not two-factor authentication are a deleted WordPress account.  It's possible that's not the case and we're directing them to login, which may be incorrect and confusing to the user.  If we can ensure it's a deleted account, we should catch that case, show an appropriate message, and track that event specifically.  If we can't, let's keep the generic error message and remove the `SignupGoogleFragment` to fix the indefinite dialog scenario.

Also, we're triggering the `trackSignupSocialAccountsNeedConnecting` statistic from the `handleUserExists` method, which should only be called when we're certain the user exists and their WordPress/Google accounts need to be connected.  That's not true if they deleted their account.

Reply:
I don't think we are making ` the assumption all scenarios that aren't an error and not two-factor authentication are a deleted WordPress account`. We are making an assumption that all scenarios that aren't an error or not two-factor authentication are successful. I've renamed the `handleUserExists` to `loginViaSocialAccount` to make it more clear. Wdyt?